fvtest/compilertriltest/IficmpTest.cpp

```diff
@@ -0,0 +1,304 @@
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+#include <gtest/gtest.h>
+#include "Jit.hpp"
+#include "JitTest.hpp"
+#include "jitbuilder_compiler.hpp"
+
+using IfcmpReductionParamType = std::tuple<std::string, std::string, std::string, std::string>;
+
+class IfcmpReductionTest : public ::testing::TestWithParam<IfcmpReductionParamType>
+   {
+   public:
+
+   static void SetUpTestCase()
+      {
+      // Disable global value propagation and local common subexpression elimination.
+      // Otherwise, the trees are optimized into nothing.
+      const char *options = "-Xjit:acceptHugeMethods,enableBasicBlockHoisting,omitFramePointer,"
+         "useIlValidator,paranoidoptcheck,disableGlobalVP,disableLocalCSE";
+
+      auto initSuccess = initializeJitWithOptions(const_cast<char*>(options));
+      ASSERT_TRUE(initSuccess) << "Failed to initialize the JIT.";
+      }
+
+   static void TearDownTestCase()
+      {
+      shutdownJit();
+      }
+   };
+
+/**
+ * @brief Struct equivalent to the IfcmpReductionParamType tuple
+ *
+ * Used for easier unpacking of test argument.
+ */
+struct IfcmpReductionParamStruct
+   {
+   std::string type;
+   std::string sign;
+   std::string val1;
+   std::string val2;
+   };
+
+/**
+ * @brief Given an instance of IfcmpReductionParamType, returns an equivalent instance
+ *    of IfcmpReductionParamStruct
+ */
+IfcmpReductionParamStruct to_struct(IfcmpReductionParamType param)
+   {
+   IfcmpReductionParamStruct s;
+   s.type = std::get<0>(param);
+   s.sign = std::get<1>(param);
+   s.val1 = std::get<2>(param);
+   s.val2 = std::get<3>(param);
+   return s;
+   }
+
+/*
+ * Test reductions of the ifcmpge opcode to ifcmpeq
+ *
+ * Currently, we cannot perform this test automatically, so we are resorted to
+ * manually running the tests and examining the trace file.
+ *
+ * Ideally, the following should be performed:
+ * 1) Construct the trees that are to be transformed
+ * 2) Do the transformation
+ * 3) Compare the resulting trees with the expected trees.
+ *
+ * However, the current implementation of Tril does not support step 3).
+ *
+ * To produce a trace log, run the following:
+ * $ TR_Options=traceilgen,tracefull,log=trace.log ./comptest --gtest_filter="IfcmpInt32Reduction/IfcmpReductionTest.Reduction/IntegerSigned_1_42"
+ *
+ * Then, open the trace log and witness the transformation.
+ */
+TEST_P(IfcmpReductionTest, Reduction)
+   {
+   auto param = to_struct(GetParam());
+
+   std::string type = param.type;
+   std::string sign = param.sign;
+   std::string val1 = param.val1;
+   std::string val2 = param.val2;
```

Comment:
I suppose this works, though I would have probably just used `param.*` everywhere :smile: .